common/option-model.cpp

```diff
@@ -0,0 +1,569 @@
+        {
+            float tmp_value = value;
+            float temp_value_displayed = tmp_value;
+            float min_range_displayed = range.min;
+            float max_range_displayed = range.max;
+
+            // computing the number of decimal digits taken from the step options' property
+            // this will then be used to format the displayed value
+            auto num_of_decimal_digits = []( float f ) {
+                float f_0 = std::fabs( f - (int)f );
+                std::string s = std::to_string( f_0 );
+                size_t cur_len = s.length();
+                // removing trailing zeros
+                while( cur_len > 3 && s[cur_len - 1] == '0' )
+                    cur_len--;
+                return cur_len - 2;
+            };
+            int num_of_decimal_digits_displayed = (int)num_of_decimal_digits( range.step );
+
+            // displaying in cm instead of meters for D405
+            if( use_cm_units )
+            {
+                temp_value_displayed *= 100.f;
+                min_range_displayed *= 100.f;
+                max_range_displayed *= 100.f;
+                int updated_num_of_decimal_digits_displayed = num_of_decimal_digits_displayed - 2;
+                if( updated_num_of_decimal_digits_displayed > 0 )
+                    num_of_decimal_digits_displayed = updated_num_of_decimal_digits_displayed;
+            }
+
+            std::stringstream formatting_ss;
+            formatting_ss << "%." << num_of_decimal_digits_displayed << "f";
+
+
+            if( ImGui::SliderFloat( id.c_str(),
+                                    &temp_value_displayed,
+                                    min_range_displayed,
+                                    max_range_displayed,
+                                    formatting_ss.str().c_str() ) )
+            {
+                tmp_value = use_cm_units ? temp_value_displayed / 100.f : temp_value_displayed;
+                auto loffset = std::abs( fmod( tmp_value, range.step ) );
+                auto roffset = range.step - loffset;
+                if( tmp_value >= 0 )
+                    tmp_value = ( loffset < roffset ) ? tmp_value - loffset : tmp_value + roffset;
+                else
+                    tmp_value = ( loffset < roffset ) ? tmp_value + loffset : tmp_value - roffset;
+                tmp_value = ( tmp_value < range.min ) ? range.min : tmp_value;
+                tmp_value = ( tmp_value > range.max ) ? range.max : tmp_value;
+
+                slider_clicked = slider_selected( opt, tmp_value, error_message, model );
+            }
+            else
+            {
+                slider_clicked = slider_unselected( opt, tmp_value, error_message, model );
+            }
+        }
+    }
+    catch( const error & e )
+    {
+        error_message = error_to_string( e );
+    }
+
+    return slider_clicked;
+}
+
+bool option_model::is_checkbox() const
+{
+    return range.max == 1.0f && range.min == 0.0f && range.step == 1.0f;
```

Comment:
Is the emitter frequency shown as checkbox in the viewer (image would be great in this PR)

Reply:
It will be shown as a combo-box.

That's why I refactored the original code who draw each 0-1 values option as a checkbox, even if it is an enum.
I will attach a screenshot tomorrow..

Reply:
![image](https://user-images.githubusercontent.com/64067618/196611564-7dca3ca5-b3a7-4ef8-b4c2-343582d0bf41.png)